hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/table/HoodieSparkCopyOnWriteTable.java

```diff
@@ -264,6 +269,24 @@ public Iterator<List<WriteStatus>> handleInsert(
     return Collections.singletonList(createHandle.close()).iterator();
   }
 
+  @Override
+  public boolean supportsFileGroupReader() {
+    return true;
+  }
+
+  @Override
+  public List<WriteStatus> runCompactionUsingFileGroupReader(String instantTime,
+                                                             CompactionOperation operation,
+                                                             HoodieReaderContext readerContext,
+                                                             Configuration conf) {
+    Option<BaseKeyGenerator> keyGeneratorOpt = HoodieSparkKeyGeneratorFactory.createBaseKeyGenerator(config);
+    HoodieSparkMergeHandleV2 mergeHandle = new HoodieSparkMergeHandleV2(config,
+        instantTime, this, operation, taskContextSupplier, keyGeneratorOpt, readerContext, conf);
+    mergeHandle.write();
+    mergeHandle.close();
```

Comment:
close() returns a List<WriteStatus> already. we don't need to make another call at L 287

Reply:
Fixed.